packages/app/src/domain/vaccine/logic/get-vaccine-coverage-display-values.ts

```diff
@@ -0,0 +1,74 @@
+import { assert } from '@corona-dashboard/common';
+import { isDefined, isPresent } from 'ts-is-present';
+import { ChoroplethDataItem, MapType } from '~/components/choropleth/logic';
+import { replaceVariablesInText } from '~/utils/replace-variables-in-text';
+import { parseFullyVaccinatedPercentageLabel } from './parse-fully-vaccinated-percentage-label';
+
+export function getVaccineCoverageDisplayValues(
+  d: ChoroplethDataItem,
+  text: Record<string, Record<string, Record<string, string>>>,
+  map: MapType,
+  formatPercentage: (value: number) => string
+) {
+  const metrics = {
+    fully_vaccinated_percentage: '–',
+    has_one_shot_percentage: '–',
+  };
+
+  const getFormattedLabelOrValue = getLabelOrValueFormatter(
+    text,
+    map,
+    formatPercentage
+  );
+
+  if ('fully_vaccinated_percentage_label' in d) {
+    metrics.fully_vaccinated_percentage = getFormattedLabelOrValue(
+      d,
+      'fully_vaccinated_percentage',
+      'fully_vaccinated_percentage_label'
+    );
+  }
+
+  if ('has_one_shot_percentage_label' in d) {
+    metrics.has_one_shot_percentage = getFormattedLabelOrValue(
+      d,
+      'has_one_shot_percentage',
+      'has_one_shot_percentage_label'
+    );
+  }
+
+  return metrics;
+}
+
+function getLabelOrValueFormatter(
+  text: Record<string, Record<string, Record<string, string>>>,
+  map: MapType,
+  formatPercentage: (value: number) => string
+) {
+  return function <T extends ChoroplethDataItem>(
+    data: T,
+    property: keyof T,
+    labelKey: keyof T
+  ) {
+    const parsedValue = isPresent(data[labelKey])
+      ? parseFullyVaccinatedPercentageLabel(data[labelKey] as unknown as string)
+      : (data[property] as unknown as undefined | number | null) || null;
```

Comment:
```suggestion
      : (data[property] as unknown as undefined | number | null) ?? null;
```
using the `||` operator would assign null in the case where data[property] returns a value of zero (0), because that is considered falseish